gdb/riscv-tdep.c

```diff
@@ -75,17 +75,19 @@ struct riscv_frame_cache
   struct trad_frame_saved_reg *saved_regs;
 };
 
+/* Use the ABI names here because Eclipse can't display names besides what we
+ * declare here, and users want to see the ABI names in any case. */
 static const char * const riscv_gdb_reg_names[RISCV_LAST_FP_REGNUM + 1] =
 {
-  "x0",  "x1",  "x2",  "x3",  "x4",  "x5",  "x6",  "x7",
-  "x8",  "x9",  "x10", "x11", "x12", "x13", "x14", "x15",
-  "x16", "x17", "x18", "x19", "x20", "x21", "x22", "x23",
-  "x24", "x25", "x26", "x27", "x28", "x29", "x30", "x31",
+  "zero", "ra", "sp", "gp", "tp", "t0", "t1", "t2",
+  "fp", "s1", "a0", "a1", "a2", "a3", "a4", "a5",
```

Comment:
I checked the disassembled code and, at least in my environment, it uses `s0`, not `fp`:

```
40400196:	1141                	addi	sp,sp,-16
40400198:	c622                	sw	s0,12(sp)
4040019a:	0800                	addi	s0,sp,16
```

Reply:
Indeed, s0 is the preferred alias for x8.